openwisp_controller/connection/tests/test_models.py

```diff
@@ -256,6 +258,22 @@ def _exec_command_return_value(
         stderr_.read().decode('utf8').strip.return_value = stderr
         return (stdin_, stdout_, stderr_)
 
+    def _run_commit_hooks(self):
```

Comment:
TransactionTestCase was still failing and also, according to [django docs](https://docs.djangoproject.com/en/3.1/topics/testing/overview/#rollback-emulation), TransactionTestCase will make our test cases approximately 3 times slower.
So i think using TestCase and clearing on_commit_hooks will be the best way for us to go for now.

Reply:
In some cases, some slowness is a trade off for a testing environment which resembles what we run in prod.
We only need to move the tests that need transactions to the TransactionsTestCase

Why is it failing?
We are using it successfully here: https://github.com/openwisp/openwisp-firmware-upgrader/blob/master/openwisp_firmware_upgrader/tests/test_openwrt_upgrader.py#L60

If it's failing, it means there are problems with the implementation which may happen in prod and it cannot be deployed.